userspace/otpilot/src/flash.rs

```diff
@@ -0,0 +1,174 @@
+// Copyright 2021 lowRISC contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use core::cell::Cell;
+
+use libtock::result::TockResult;
+use libtock::syscalls;
+use libtock::syscalls::raw::yieldk;
+
+pub const MAX_BUFFER_LENGTH: usize = 128;
+
+pub trait Flash {
+    // Read from flash.
+    // offset: Location relative to flash start to read from. Must be word (4 bytes) aligned.
+    // buffer: Buffer for read data.
+    //         It must be: buffer.len() <= len <= MAX_BUFFER_LENGTH
+    // read_length: Number of bytes to read. Must be > 0 and a multiple of 4.
+    fn read(&self, offset: usize, buffer: &mut[u8], len: usize) -> TockResult<()>;
+
+    // Write to flash.
+    // offset: Location relative to flash start to write to. Must be word (4 bytes) aligned.
+    // buffer: Buffer with data to write.
+    //         It must be: buffer.len() <= len <= MAX_BUFFER_LENGTH
+    // read_length: Number of bytes to write. Must be > 0 and a multiple of 4.
```

Comment:
This should be `write_length`, not `read_length`.  And, again, the parameter name (`len`) is inconsistent with this comment.

Reply:
Fixed.